docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+
+Rather than having to specify the same values for colors or fonts in every widget's CSS module, abstract custom properties can instead be referenced by name, with values then provided in a centralized CSS `:root` pseudo-class. This separation allows for much simpler maintenance of common styling concerns across an entire application.
+
+Note that the `:root` pseudo-class is global within a webpage, but through Dojo's use of CSS modules, `:root` properties could potentially be specified in many locations across an application. However, Dojo does not guarantee the order in which CSS modules are processed, so to ensure consistency of which properties appear in `:root`, it is recommended that applications use a single `:root` definition within a centralized `variables.css` file in the application's codebase. This centralized variables file is a regular CSS file (not a CSS module) and can be `@import`ed as such in any CSS modules that require custom property values.
+
+For example:
+
+> src/themes/variables.css
+
+```css
+:root {
+	/* different sets of custom properties can be used if an application supports more than one possible theme */
+	--light-background: lightgray;
+	--light-foreground: black;
+
+	--dark-background: black;
+	--dark-foreground: lightgray;
+
+	--font: monospace;
+	--padding: 32px;
+}
+```
+
+> src/themes/myDarkTheme/MyWidget.m.css
+
+```css
+@import '../variables.css';
+
+.root {
+	font-family: var(--font);
+	margin: var(--padding);
+
+	color: var(--dark-foreground);
+	background: var(--dark-background);
+}
+```
+
+### CSS module composition
+
+Applying a theme to a Dojo widget results in the widget's default styling classes being entirely overridden by those provided in the theme (see [Note 3 of `ThemedMixin`'s `this.theme()` method](#themedmixin-theme-method) for details). This can be problematic when only a subset of properties in a given styling class need to be modified through a theme, while the remainder can stay as default.
+
+[CSS module files](https://github.com/css-modules/css-modules) in Dojo applications can leverage `composes:` functionality to apply sets of styles from one class selector to another. This can be useful when creating a new theme that tweaks an existing one, as well as for general abstraction of common sets of styling properties within a single theme (note that [CSS custom properties](#css-custom-properties) are a more standardized way of abstracting values for individual style properties).
+
+**Warning:** Use of `composes:` can prove brittle, for example when extending a third-party theme that is not under direct control of the current application. Any change made by a third-party could break an application theme that `composes` the underlying theme, and such breakages can be problematic to pin down and resolve.
+
+However, careful use of this feature can be helpful in large applications. For example, centralizing a common set of properties:
+
+> src/themes/common/ButtonArchetype.m.css
```

Comment:
Shall we use `base` over `archetype`? I just think it will be more obvious to a wider audience.